src/main/java/ui/components/pickers/PickerAssignee.java

```diff
@@ -0,0 +1,173 @@
+package ui.components.pickers;
+
+import backend.resource.TurboUser;
+import com.sun.javafx.tk.FontLoader;
+import com.sun.javafx.tk.Toolkit;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+import java.util.List;
+import java.util.Optional;
+
+public class PickerAssignee extends TurboUser implements Comparable<PickerAssignee> {
+
+    private static final int AVATAR_SIZE = 30;
+    private static final int LABEL_HEIGHT = 40;
+
+    private boolean isExisting = false;
+    private boolean isSelected = false;
+    // set to true so that all users are matching when the text field is empty
+    private boolean isMatching = true;
+
+    public PickerAssignee(TurboUser user) {
+        super(user);
+    }
+
+    public PickerAssignee(PickerAssignee assignee) {
+        this((TurboUser) assignee);
+        setExisting(assignee.isExisting());
+        setSelected(assignee.isSelected());
+        setMatching(assignee.isMatching());
+    }
+
+    public Node getMatchingNode() {
+        HBox assigneeBox = getAssigneeHBox();
+        ImageView avatar = getAvatarImageView();
+        Label assigneeLoginName = getAssigneeLabel();
+
+        if (isSelected) {
+            assigneeLoginName.setText(assigneeLoginName.getText() + " ✓");
+            assigneeBox.setStyle(assigneeBox.getStyle() + "-fx-background-color: skyblue");
+        }
+
+        assigneeBox.getChildren().setAll(avatar, assigneeLoginName);
+
+        return assigneeBox;
+    }
+
+    public Node getNewlyAssignedAssigneeNode() {
+        return getAssigneeLabelWithAvatar();
+    }
+
+    public Node getExistingAssigneeNode(boolean hasSelected) {
+        Label assignee = getAssigneeLabelWithAvatar();
+        if (hasSelected || !isSelected()) {
```

Comment:
This part is a little unclear. What does each one refer to? Maybe make the names a bit more specific.